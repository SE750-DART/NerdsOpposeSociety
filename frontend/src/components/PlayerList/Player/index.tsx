import React from "react";
import styles from "./style.module.css";

interface PlayerProps {
  nickname: string;
  score?: number;
  highlight?: boolean;
  divider: boolean;
  isHost?: boolean;
}

const Player = ({
  nickname,
  score,
  highlight = false,
  divider,
  isHost,
}: PlayerProps) => {
  let highlightStyle = "";
  if (highlight) {
    highlightStyle = styles.playerMe;
  }

  return (
    <div className={`${styles.player} ${highlightStyle}`}>
      <div className={styles.playerText}>{nickname}</div>
      {isHost && <div className={styles.playerIndicator}>HOST</div>}
      <div className={styles.playerScore}>{score}</div>
      {divider && <div className={styles.playerDivider} />}
    </div>
  );
};

export default Player;
