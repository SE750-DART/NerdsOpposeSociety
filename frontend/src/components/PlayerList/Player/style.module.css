.player {
  align-items: center;
  display: grid;
  font-size: 24px;
  grid-template-columns: auto 56px 56px;
  grid-template-rows: 54px 2px;
  height: 56px;
  width: 100%;
}

/*
 * I'm not sure why, but CSS Modules snake-case selection seems to have broken in here. Grr
 */

.playerText {
  margin-left: 17px;
  overflow: hidden;
  text-align: left;
  text-overflow: ellipsis;
}

.playerIndicator {
  font-size: 16px;
  font-weight: bold;
  grid-column: 2;
}

.playerScore {
  grid-column: 3;
}

.playerDivider {
  background-color: #84b5de;
  grid-column: 1 / 4;
  grid-row: 2;
  height: 2px;
  width: 100%;
}

.playerMe {
  background-color: #eeffff;
  border-radius: 16px;
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.1);
}
