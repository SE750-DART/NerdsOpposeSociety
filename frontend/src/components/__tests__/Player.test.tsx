import React from "react";
import renderer from "react-test-renderer";
import Player from "../PlayerList/Player";

it("Player snapshot testing", () => {
  const treeBasicPlayer = renderer
    .create(<Player nickname="Di Kun Ong" divider={false} score={4} />)
    .toJSON();
  expect(treeBasicPlayer).toMatchSnapshot();

  const treeDivider = renderer
    .create(<Player nickname="Alex Verkerk" divider />)
    .toJSON();
  expect(treeDivider).toMatchSnapshot();

  const treeHost = renderer
    .create(<Player nickname="Rawiri Hohepa" divider={false} isHost />)
    .toJSON();
  expect(treeHost).toMatchSnapshot();

  const treeIsMe = renderer
    .create(
      <Player nickname="Tait Fuller" divider={false} highlight score={3} />
    )
    .toJSON();
  expect(treeIsMe).toMatchSnapshot();
});
