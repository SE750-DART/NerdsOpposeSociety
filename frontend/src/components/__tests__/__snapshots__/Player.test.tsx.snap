// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player snapshot testing 1`] = `
<div
  className="player "
>
  <div
    className="playerText"
  >
    Di Kun Ong
  </div>
  <div
    className="playerScore"
  >
    4
  </div>
</div>
`;

exports[`Player snapshot testing 2`] = `
<div
  className="player "
>
  <div
    className="playerText"
  >
    Alex Verkerk
  </div>
  <div
    className="playerScore"
  />
  <div
    className="playerDivider"
  />
</div>
`;

exports[`Player snapshot testing 3`] = `
<div
  className="player "
>
  <div
    className="playerText"
  >
    Rawiri Hohepa
  </div>
  <div
    className="playerIndicator"
  >
    HOST
  </div>
  <div
    className="playerScore"
  />
</div>
`;

exports[`Player snapshot testing 4`] = `
<div
  className="player playerMe"
>
  <div
    className="playerText"
  >
    Tait Fuller
  </div>
  <div
    className="playerScore"
  >
    3
  </div>
</div>
`;
